beacon/state.go

```diff
@@ -29,16 +29,18 @@ type state struct {
 	proposalChecker           eligibilityChecker
 }
 
-func newState(logger log.Log, cfg Config, epochWeight uint64, atxs []types.ATXID) *state {
+type checkerFunc func(log.Log, Config, int) eligibilityChecker
+
+func newState(logger log.Log, cfg Config, epochWeight uint64, atxids []types.ATXID, create checkerFunc) *state {
 	return &state{
 		logger:                  logger,
 		epochWeight:             epochWeight,
-		atxs:                    atxs,
+		atxs:                    atxids,
 		firstRoundIncomingVotes: make(map[string]proposalList),
 		votesMargin:             map[string]*big.Int{},
 		hasProposed:             make(map[string]struct{}),
 		hasVoted:                make([]map[string]struct{}, cfg.RoundsNumber),
-		proposalChecker:         createProposalChecker(logger, cfg.Kappa, cfg.Q, epochWeight),
+		proposalChecker:         create(logger, cfg, len(atxids)),
```

Comment:
is it possible to pass already created object?

Reply:
turned out i don't really need to manipulate the checker. removed